src/FffGcodeWriter.cpp

```diff
@@ -499,6 +499,17 @@ void FffGcodeWriter::processStartingCode(const SliceDataStorage& storage, const
     {
         gcode.writeExtrusionMode(true);
     }
+    if (gcode.getFlavor() != EGCodeFlavor::GRIFFIN)
+    {
+        // ensure extruder is zeroed
+        gcode.resetExtrusionValue();
```

Comment:
I suggest to execute this line only if the retraction is going to be made. In this case, if the retraction is not enabled, the behavior is the same as before.

Reply:
Yes, but as I mention above there doesn't appear to be any other G92 output to reset the extruder. If no G92 is output then it assumes the extruder is zeroed by some other means - is this safe? I agree that is how it is currently working so I don't have a big problem with your suggestion.

Reply:
OK, I did as you asked.